packages/components/src/search-control/index.native.js

```diff
@@ -1,96 +1,235 @@
+	close as closeIcon,
 } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import styles from './style.scss';
+import baseStyles from './style.scss';
+import platformStyles from './platform-style.scss';
+
+// Merge platform specific styles
+for ( const selector in platformStyles ) {
+	baseStyles[ selector ] = {
+		...baseStyles[ selector ],
+		...platformStyles[ selector ],
+	};
+}
+
+function selectModifiedStyles( styles, modifier ) {
+	const modifierMatcher = new RegExp( `--${ modifier }$` );
+	const modifierSelectors = Object.keys( styles ).filter( ( selector ) =>
+		selector.match( modifierMatcher )
+	);
+
+	return modifierSelectors.reduce( ( modifiedStyles, modifierSelector ) => {
+		const blockElementSelector = modifierSelector.split( '--' )[ 0 ];
+		modifiedStyles[ blockElementSelector ] = styles[ modifierSelector ];
+		return modifiedStyles;
+	}, {} );
+}
+
+function mergeStyles( styles, updateStyles, selectors ) {
+	selectors.forEach( ( selector ) => {
+		styles[ selector ] = {
+			...styles[ selector ],
+			...updateStyles[ selector ],
+		};
+	} );
+
+	return styles;
+}
 
 function SearchControl( {
 	value,
 	onChange,
-	label,
-	placeholder = __( 'Search' ),
+	placeholder = __( 'Search Blocks' ),
 } ) {
 	const [ isActive, setIsActive ] = useState( false );
+	const [ currentStyles, setCurrentStyles ] = useState( baseStyles );
 
+	const isDark = useColorScheme() === 'dark';
 	const inputRef = useRef();
 
-	const searchFormStyle = usePreferredColorSchemeStyle(
-		styles.searchForm,
-		styles.searchFormDark
-	);
+	const isIOS = Platform.OS === 'ios';
 
-	const searchFormInputStyle = usePreferredColorSchemeStyle(
-		styles.searchFormInput,
-		styles.searchFormInputDark
-	);
+	const darkStyles = useMemo( () => {
+		return selectModifiedStyles( baseStyles, 'dark' );
+	}, [] );
 
-	const placeholderStyle = usePreferredColorSchemeStyle(
-		styles.searchFormPlaceholder,
-		styles.searchFormPlaceholderDark
-	);
+	const activeStyles = useMemo( () => {
+		return selectModifiedStyles( baseStyles, 'active' );
+	}, [] );
 
-	return (
-		<TouchableHighlight accessible={ false }>
-			<View style={ searchFormStyle }>
-				{ isActive ? (
-					<ToolbarButton
-						title={ __( 'Cancel search' ) }
-						icon={ arrowLeft }
-						onClick={ () => {
-							inputRef.current.blur();
-							onChange( '' );
-							setIsActive( false );
-						} }
-					/>
-				) : (
-					<ToolbarButton
-						title={ label }
-						icon={ searchIcon }
-						onClick={ () => {
-							inputRef.current.focus();
-							setIsActive( true );
-						} }
-					/>
-				) }
-				<TextInput
-					ref={ inputRef }
-					style={ searchFormInputStyle }
-					placeholderTextColor={ placeholderStyle.color }
-					onChangeText={ onChange }
-					onFocus={ () => setIsActive( true ) }
-					value={ value }
-					placeholder={ placeholder }
+	const activeDarkStyles = useMemo( () => {
+		return selectModifiedStyles( baseStyles, 'active-dark' );
+	}, [] );
+
+	useEffect( () => {
```

Comment:
The idea here is to use a  hook to build the styles on demand given any state changes. It avoids having to weave conditionals in the components per state e.g. `<someComponent style = { isActive ? activeStyle : style } />` . This approach can also be used to replace the `usePreferredColorSchemeStyle` which needs be called on every individual set of style rules. Instead one single call can be made on the entire style object to generate the themed styles.

I'd like to extract this way of handling the styles out to a hook since it's a more declarative way to build the style object for the component. In the interest of time however, I opted to leave this idea scoped to this module.